node/src/accountant/payment_adjuster/preparatory_analyser/accounts_abstraction.rs

```diff
@@ -0,0 +1,70 @@
+use crate::accountant::payment_adjuster::disqualification_arbiter::DisqualificationArbiter;
+use crate::accountant::payment_adjuster::miscellaneous::data_structures::WeightedPayable;
+use crate::accountant::{AnalyzedPayableAccount, QualifiedPayableAccount};
+
+pub trait DisqualificationAnalysableAccount<Product>: BalanceProvidingAccount
+where
+    Product: BalanceProvidingAccount + DisqualificationLimitProvidingAccount,
+{
+    fn prepare_analyzable_account(
+        self,
+        disqualification_arbiter: &DisqualificationArbiter,
+    ) -> Product;
+}
+
+pub trait BalanceProvidingAccount {
+    fn balance_minor(&self) -> u128;
+}
+
+pub trait DisqualificationLimitProvidingAccount {
+    fn disqualification_limit(&self) -> u128;
+}
```

Comment:
If you can combine the trait, it'll be more concise:

So, if you think generic name is the only limitation, then this is the name we both came up with: `AccountWithBalanceAndDisqLimit`